MacOS Py3x Functional+Single job failing for PR Validation pipelines (AzDO)

See https://dev.azure.com/ms/vscode-python/_build/results?buildId=118775&view=logs&j=0e2bd8aa-634b-5e2a-1491-c038c1dc18c7&t=9f8541d0-5ac5-515a-c7fb-d62f0b3bb0e4

It's the "Fix powershell problem on Ubuntu 20" step that's failing:

![image](https://user-images.githubusercontent.com/51720070/97371711-1eaca700-186f-11eb-89ab-fd467ced7c01.png)


Raw log:
```
2020-10-27T22:53:28.9813510Z ##[debug]Evaluating condition for step: 'Fix powershell problem on Ubuntu 20'
2020-10-27T22:53:28.9815960Z ##[debug]Evaluating: ne(variables['Agent.Os'], 'Windows_NT')
2020-10-27T22:53:28.9817000Z ##[debug]Evaluating ne:
2020-10-27T22:53:28.9818280Z ##[debug]..Evaluating indexer:
2020-10-27T22:53:28.9819240Z ##[debug]....Evaluating variables:
2020-10-27T22:53:28.9820150Z ##[debug]....=> Object
2020-10-27T22:53:28.9821090Z ##[debug]....Evaluating String:
2020-10-27T22:53:28.9822010Z ##[debug]....=> 'Agent.Os'
2020-10-27T22:53:28.9822990Z ##[debug]..=> 'Darwin'
2020-10-27T22:53:28.9823920Z ##[debug]..Evaluating String:
2020-10-27T22:53:28.9824830Z ##[debug]..=> 'Windows_NT'
2020-10-27T22:53:28.9825750Z ##[debug]=> True
2020-10-27T22:53:28.9827140Z ##[debug]Expanded: ne('Darwin', 'Windows_NT')
2020-10-27T22:53:28.9828420Z ##[debug]Result: True
2020-10-27T22:53:28.9829870Z ##[section]Starting: Fix powershell problem on Ubuntu 20
2020-10-27T22:53:28.9839530Z ==============================================================================
2020-10-27T22:53:28.9840560Z Task         : Bash
2020-10-27T22:53:28.9841850Z Description  : Run a Bash script on macOS, Linux, or Windows
2020-10-27T22:53:28.9843340Z Version      : 3.176.0
2020-10-27T22:53:28.9844330Z Author       : Microsoft Corporation
2020-10-27T22:53:28.9845740Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
2020-10-27T22:53:28.9846820Z ==============================================================================
2020-10-27T22:53:28.9925970Z ##[debug]Using node path: /Users/runner/runners/2.175.2/externals/node/bin/node
2020-10-27T22:53:29.1414760Z ##[debug]agent.TempDirectory=/Users/runner/work/_temp
2020-10-27T22:53:29.1441850Z ##[debug]loading inputs and endpoints
2020-10-27T22:53:29.1447890Z ##[debug]loading INPUT_TARGETTYPE
2020-10-27T22:53:29.1465010Z ##[debug]loading INPUT_FILEPATH
2020-10-27T22:53:29.1468810Z ##[debug]loading INPUT_SCRIPT
2020-10-27T22:53:29.1470150Z ##[debug]loading INPUT_WORKINGDIRECTORY
2020-10-27T22:53:29.1471500Z ##[debug]loading INPUT_FAILONSTDERR
2020-10-27T22:53:29.1472980Z ##[debug]loading INPUT_NOPROFILE
2020-10-27T22:53:29.1474400Z ##[debug]loading INPUT_NORC
2020-10-27T22:53:29.1475930Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2020-10-27T22:53:29.1477500Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2020-10-27T22:53:29.1479080Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2020-10-27T22:53:29.1488320Z ##[debug]loading SECRET_AZURE_COGNITIVE_KEY
2020-10-27T22:53:29.1489420Z ##[debug]loading SECRET_AZURESUBSCRIPTIONFORBLOBUPLOAD
2020-10-27T22:53:29.1490500Z ##[debug]loading SECRET_AZURE_SUBSCRIPTION
2020-10-27T22:53:29.1491440Z ##[debug]loading SECRET_VSCE_PAT
2020-10-27T22:53:29.1492410Z ##[debug]loading SECRET_CODECOV_TOKEN
2020-10-27T22:53:29.1493560Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2020-10-27T22:53:29.1494720Z ##[debug]loading SECRET_GITHUB_TOKEN
2020-10-27T22:53:29.1495710Z ##[debug]loading SECRET_AZURE_STORAGE_ACCESS_KEY
2020-10-27T22:53:29.1496880Z ##[debug]loading SECRET_ARTIFACTFEED
2020-10-27T22:53:29.1508050Z ##[debug]loading SECRET_COV_UUID
2020-10-27T22:53:29.1509570Z ##[debug]loading SECRET_COVERALLS_REPO_TOKEN
2020-10-27T22:53:29.1511170Z ##[debug]loading SECRET_AZURE_COGNITIVE_ENDPOINT
2020-10-27T22:53:29.1512530Z ##[debug]loaded 22
2020-10-27T22:53:29.1532430Z ##[debug]Agent.ProxyUrl=undefined
2020-10-27T22:53:29.1533810Z ##[debug]Agent.CAInfo=undefined
2020-10-27T22:53:29.1535140Z ##[debug]Agent.ClientCert=undefined
2020-10-27T22:53:29.1536200Z ##[debug]Agent.SkipCertValidation=undefined
2020-10-27T22:53:29.1547280Z ##[debug]noProfile=true
2020-10-27T22:53:29.1553270Z ##[debug]noRc=true
2020-10-27T22:53:29.1573080Z ##[debug]check path : /Users/runner/work/_tasks/Bash_6c731c3c-3c68-459a-a5c9-bde6e6595b5b/3.176.0/task.json
2020-10-27T22:53:29.1574580Z ##[debug]adding resource file: /Users/runner/work/_tasks/Bash_6c731c3c-3c68-459a-a5c9-bde6e6595b5b/3.176.0/task.json
2020-10-27T22:53:29.1575770Z ##[debug]system.culture=en-US
2020-10-27T22:53:29.1593030Z ##[debug]failOnStderr=false
2020-10-27T22:53:29.1595700Z ##[debug]workingDirectory=/Users/runner/work/1/s
2020-10-27T22:53:29.1596470Z ##[debug]check path : /Users/runner/work/1/s
2020-10-27T22:53:29.1598140Z ##[debug]targetType=inline
2020-10-27T22:53:29.1600660Z ##[debug]script=sudo chmod -R 777 /run/user
2020-10-27T22:53:29.1647930Z Generating script.
2020-10-27T22:53:29.1650850Z ##[debug]which 'bash'
2020-10-27T22:53:29.1665230Z ##[debug]found: '/bin/bash'
2020-10-27T22:53:29.1666050Z Script contents:
2020-10-27T22:53:29.1667340Z sudo chmod -R 777 /run/user
2020-10-27T22:53:29.1668550Z ##[debug]Agent.Version=2.175.2
2020-10-27T22:53:29.1669550Z ##[debug]agent.tempDirectory=/Users/runner/work/_temp
2020-10-27T22:53:29.1670550Z ##[debug]check path : /Users/runner/work/_temp
2020-10-27T22:53:29.1695330Z ========================== Starting Command Output ===========================
2020-10-27T22:53:29.1697450Z ##[debug]which '/bin/bash'
2020-10-27T22:53:29.1698470Z ##[debug]found: '/bin/bash'
2020-10-27T22:53:29.1699470Z ##[debug]/bin/bash arg: --noprofile
2020-10-27T22:53:29.1700450Z ##[debug]/bin/bash arg: --norc
2020-10-27T22:53:29.1702030Z ##[debug]/bin/bash arg: /Users/runner/work/_temp/3a4a4b2d-c5fe-4ebd-ac8f-6fccb82f99f6.sh
2020-10-27T22:53:29.1703550Z ##[debug]exec tool: /bin/bash
2020-10-27T22:53:29.1704500Z ##[debug]arguments:
2020-10-27T22:53:29.1705740Z ##[debug]   --noprofile
2020-10-27T22:53:29.1706920Z ##[debug]   --norc
2020-10-27T22:53:29.1708260Z ##[debug]   /Users/runner/work/_temp/3a4a4b2d-c5fe-4ebd-ac8f-6fccb82f99f6.sh
2020-10-27T22:53:29.1710080Z [command]/bin/bash --noprofile --norc /Users/runner/work/_temp/3a4a4b2d-c5fe-4ebd-ac8f-6fccb82f99f6.sh
2020-10-27T22:53:29.2401270Z chmod: /run/user: No such file or directory
2020-10-27T22:53:29.2435860Z 
2020-10-27T22:53:29.2446120Z ##[debug]Exit code 1 received from tool '/bin/bash'
2020-10-27T22:53:29.2459280Z ##[debug]STDIO streams have closed for tool '/bin/bash'
2020-10-27T22:53:29.2526740Z ##[error]Bash exited with code '1'.
2020-10-27T22:53:29.2538260Z ##[debug]Processed: ##vso[task.issue type=error;]Bash exited with code '1'.
2020-10-27T22:53:29.2539440Z ##[debug]task result: Failed
2020-10-27T22:53:29.2542470Z ##[debug]Processed: ##vso[task.complete result=Failed;done=true;]
2020-10-27T22:53:29.2546530Z ##[section]Finishing: Fix powershell problem on Ubuntu 20
```

--- a/build/ci/templates/test_phases.yml
+++ b/build/ci/templates/test_phases.yml
@@ -465,7 +465,7 @@ steps:
   - bash: |
       sudo chmod -R 777 /run/user
     displayName: 'Fix powershell problem on Ubuntu 20'
-    condition: ne(variables['Agent.Os'], 'Windows_NT')
+    condition: and(succeeded(), eq(variables['Agent.Os'], 'Linux'), or(contains(variables['TestsToRun'], 'testSmoke'), contains(variables['TestsToRun'], 'testInsiders')))
 
   # Now actually install 
   - powershell: |

